File: exercises/beecrowd/1165/1165.ts
// https://judge.beecrowd.com/pt/problems/view/1165
import * as fs from "fs";

function isPrime(num: number): boolean {
  if (num <= 1) {
    return false;
  }
  const sqrtNum: number = Math.sqrt(num);
  for (let i: number = 2; i <= sqrtNum; i++) {
    if (num % i === 0) {
      return false;
    }
  }
  return true;
}

function main(): void {
  const lines: number[] = fs
    .readFileSync("/dev/stdin", "utf8")
    .trim()
    .split("\n")
    .map(Number);

  let lineIndex: number = 0;
  const numTestCases: number = lines[lineIndex++];

  for (let i: number = 0; i < numTestCases; i++) {
    const currentNum: number = lines[lineIndex++];

    if (isPrime(currentNum)) {
      console.log(`${currentNum} eh primo`);
    } else {
      console.log(`${currentNum} nao eh primo`);
    }
  }
}

main();

File: exercises/beecrowd/1165/1165.go
// https://judge.beecrowd.com/pt/problems/view/1165
package main

import (
	"fmt"
)

func main() {
	var x int
	fmt.Scan(&x)

	for i := 0; i < x; i++ {
		var num int
		fmt.Scan(&num)

		isPrime := true
		if num <= 1 {
			isPrime = false
		} else {
			for j := 2; j*j <= num; j++ {
				if num%j == 0 {
					isPrime = false
					break
				}
			}
		}

		if isPrime {
			fmt.Printf("%d eh primo\n", num)
		} else {
			fmt.Printf("%d nao eh primo\n", num)
		}
	}
}
